python/ray/data/dataset.py

```diff
@@ -3186,6 +3186,48 @@ def write_mongo(
             collection=collection,
         )
 
+    @ConsumptionAPI
+    def write_bigquery(
+        self,
+        dataset: str,
+        project_id: str,
+        ray_remote_args: Dict[str, Any] = None,
+    ) -> None:
+        """Write the dataset to a BigQuery dataset table.
+
+        This is only supported for datasets convertible to Arrow records.
+        To control the number of parallel write tasks, use ``.repartition()``
+        before calling this method.
+
+        Examples:
+            >>> import ray
+            >>> import pandas as pd
+            >>> docs = [{"title": "BigQuery Datasource test"} for key in range(4)]
+            >>> ds = ray.data.from_pandas(pd.DataFrame(docs))
+            >>> ds.write_bigquery( # doctest: +SKIP
+            >>>     BigQueryDatasource(), # doctest: +SKIP
+            >>>     project_id="my_project_id" # doctest: +SKIP
+            >>>     dataset="my_dataset_table", # doctest: +SKIP
+            >>> ) # doctest: +SKIP
+
+        Args:
+            project_id: The name of the associated Google Cloud Project that hosts
+                the dataset to read. For more information, see details in
+                https://cloud.google.com/resource-manager/docs/creating-managing-projects.
```

Comment:
(You'd have to wrap this to appease flake8)
```suggestion
                the dataset to read. For more information, see details in
                `Creating and managing projects <https://cloud.google.com/resource-manager/docs/creating-managing-projects>`.
```

Reply:
Done